src/bp/ui-studio/src/web/components/App/index.tsx

```diff
@@ -112,36 +109,15 @@ class App extends Component<Props> {
         history.push(payload)
       }
     })
-
-    this.interval = setInterval(async () => {
-      await this.tryRefreshToken()
-    }, REFRESH_INTERVAL)
-  }
-
-  async tryRefreshToken() {
-    try {
-      if (!tokenNeedsRefresh()) {
-        return
-      }
-
-      const tokenData = getToken(false) as StoredToken
-
-      const { data } = await axios.get(`${window.API_PATH}/auth/refresh`)
-      const { newToken } = data.payload
-
-      if (newToken !== tokenData.token) {
-        setToken(newToken)
-        console.info('Token refreshed successfully')
-      } else {
-        clearInterval(this.interval)
-      }
-    } catch (err) {
-      console.error('Error validating & refreshing token', err)
-    }
   }
 
   render() {
-    return routes()
+    return (
+      <Fragment>
+        <TokenRefresher getAxiosClient={() => axios} onRefreshCompleted={token => setToken(token)} />
```

Comment:
again my opinion is simply that I feel this weird ... the whole TokenRefresher thing could be a simple javascript function to which you provide a callback or whatever ..

Reply:
I just wanted to keep it simple so it's a standalone component, with its state and everything, I agree it's not like other components, but i'll keep a note on my bucket list to change that